### PR TITLE
Add StandardSplitToggleChip

### PR DESCRIPTION
**Pull request by DenoRS**
#### WHAT

- Added StandardSplitToggleChip component

#### WHY

- resolves https://github.com/google/horologist/issues/1318

#### HOW

- Implemented StandardSplitToggleChip, similar to StandardToggleChip
- Added previews to debug folder
- Added snapshot tests

#### Screenshots

**Variants preview**:

image

Overflow preview:

image

#### Checklist :clipboard:
- [x] Add explicit visibility modifier and explicit return types for public declarations
- [x] Run spotless check
- [x] Run tests
- [x] Update metalava's signature text files

### CODE REVIEW

File: base-ui/src/test/java/com/google/android/horologist/base/ui/components/StandardSplitToggleChipTest.kt [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+    @Test
+    fun withSecondaryLabelAndIcon() {
+        screenshotTestRule.setContent(isComponent = true, takeScreenshot = true) {
+            StandardSplitToggleChip(
+                checked = true,
+                onCheckedChanged = { },
+                label = "Primary label",
+                onClick = { },
+                toggleControl = StandardSplitToggleChipToggleControl.Switch,
+                secondaryLabel = "Secondary label"
```
Comment (luizgrp) [inline base-ui/src/test/java/com/google/android/horologist/base/ui/components/StandardSplitToggleChipTest.kt:100]: nit: given this component does not have icon, this test can be removed, it is the same as the one above

File: base-ui/src/test/snapshots/images/com.google.android.horologist.base.ui.components_StandardSplitToggleChipTest_withSecondaryLabelAndIcon.png
Comment (luizgrp) [inline base-ui/src/test/snapshots/images/com.google.android.horologist.base.ui.components_StandardSplitToggleChipTest_withSecondaryLabelAndIcon.png:1]: file can be removed when test is removed

File: base-ui/src/main/java/com/google/android/horologist/base/ui/components/StandardSplitToggleChip.kt [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        colors = colors,
+        enabled = enabled,
+        checkedInteractionSource = checkedInteractionSource,
+        clickInteractionSource = clickInteractionSource,
+        contentPadding = contentPadding,
+        shape = shape
+    )
+}
+
+public enum class StandardSplitToggleChipToggleControl {
```
Comment (luizgrp) [inline base-ui/src/main/java/com/google/android/horologist/base/ui/components/StandardSplitToggleChip.kt:128]: nit: I would use the same enum for both StandardToggleChip and StandardSplitToggleChip, so I'd probably extract the enum class to its own file - happy to merge as it is and can address it in a separate pr

**Review by luizgrp — COMMENTED** [review]
Thank you!

Could you please update the branch with latest in main to pick up https://github.com/google/horologist/pull/1334 so that we can download and see the diff errors reported in CI?

File: base-ui/src/test/snapshots/images/com.google.android.horologist.base.ui.components_StandardSplitToggleChipTest_withSecondaryLabelAndIcon.png
Comment (DenoRS) [reply base-ui/src/test/snapshots/images/com.google.android.horologist.base.ui.components_StandardSplitToggleChipTest_withSecondaryLabelAndIcon.png:1]: removed

File: base-ui/src/test/java/com/google/android/horologist/base/ui/components/StandardSplitToggleChipTest.kt (hunk 1 above)
Comment (DenoRS) [reply base-ui/src/test/java/com/google/android/horologist/base/ui/components/StandardSplitToggleChipTest.kt:87]: thanks, removed

File: base-ui/src/main/java/com/google/android/horologist/base/ui/components/StandardSplitToggleChip.kt (hunk 2 above)
Comment (DenoRS) [reply base-ui/src/main/java/com/google/android/horologist/base/ui/components/StandardSplitToggleChip.kt]: Good idea! I moved this to a common folder. Let me know if you have preference for a better name.

File: base-ui/src/main/java/com/google/android/horologist/base/ui/components/StandardSplitToggleChip.kt [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    onClick: () -> Unit,
+    toggleControl: StandardSplitToggleChipToggleControl,
+    modifier: Modifier = Modifier,
+    secondaryLabel: String? = null,
+    colors: SplitToggleChipColors = ToggleChipDefaults.splitToggleChipColors(),
+    enabled: Boolean = true,
+    checkedInteractionSource: MutableInteractionSource = remember { MutableInteractionSource() },
+    clickInteractionSource: MutableInteractionSource = remember { MutableInteractionSource() },
+    contentPadding: PaddingValues = ToggleChipDefaults.ContentPadding,
+    shape: Shape = MaterialTheme.shapes.small
```
[the remark this answers is not in the document]
Comment (DenoRS) [reply base-ui/src/main/java/com/google/android/horologist/base/ui/components/StandardSplitToggleChip.kt]: Sounds good! removed

File: base-ui/src/test/java/com/google/android/horologist/base/ui/components/StandardSplitToggleChipTest.kt [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+ */
+
+package com.google.android.horologist.base.ui.components
+
+import com.google.accompanist.testharness.TestHarness
+import com.google.android.horologist.base.ui.common.StandardToggleChipToggleControl
+import com.google.android.horologist.screenshots.ScreenshotBaseTest
+import org.junit.Test
+
+class StandardSplitToggleChipTest : ScreenshotBaseTest() {
```
Comment (luizgrp) [inline base-ui/src/test/java/com/google/android/horologist/base/ui/components/StandardSplitToggleChipTest.kt:24]: Could you please add `@Ignore` annotation to this class?

The gradient of the images generated in CI are different:
com google android horologist base ui components_StandardSplitToggleChipTest_checkbox_delta

Once this is merged I will generate the images in a linux machine and upload.

**Review by luizgrp — COMMENTED** [review]
Could you please run `spotlessApply` gradle task to fix the failure:

```
Caused by: org.gradle.api.GradleException: The following files had format violations:
    src/main/java/com/google/android/horologist/base/ui/common/StandardToggleChipToggleControl.kt
```

and it should be good to merge 👍

**Review by luizgrp — APPROVED** [review]
LGTM!

I will merge as it is and follow up on the issue with the snapshot images, thank you!

**Pull request merged**